api/src/main/java/org/eclipse/microprofile/concurrent/ManagedExecutorConfig.java

```diff
@@ -76,6 +89,10 @@
      *
      * <p>The default value of <code>-1</code> indicates no upper bound,
      * although practically, resource constraints of the system will apply.</p>
+     *
+     * <p>A <code>ManagedExecutor</code> must fail to inject, raising
+     * <code>DefinitionException</code> on application startup, if the
+     * <code>maxQueued</code> value is 0 or less than -1.
```

Comment:
Why can't `maxQueued` be zero?

Intuitively, that would mean that if the maximum number of tasks are already executing, trying to submit a new one would result in an exception.

I note that this change is to match the doc on `ManagedExecutorBuilder`, so I can raise a new issue for this question instead if that's preferable.

Reply:
@Azquelt It would also mean that when you submit your very first task, if the system doesn't give you priority over everything thing else and start running that task immediately (if necessary even creating a new thread to do that), then your task or dependent stage fails because it cannot queue anything.  This is a bad practice that I don't want to use the spec to encourage it.  Inevitably, if 0 is an allowed value, users will try to configure it because at first glance it sounds nice (my work is really important - never queue it), and then end up with all sorts of unwanted failures in their application.  It would also disallow implementations from taking the approach of queuing every task and pulling from the queue as threads become available.  I cannot see adding a change that will lead to users following a bad practice and will be unnecessarily limiting upon implementations.

That said, I could see a case for instead allowing a 0 value for `maxAsync`.  A user might want to enforce that no one can use the *Async methods for dependent stages.  If this is something you were hoping for, I'm willing to allow the 0 value there instead.

Reply:
If we do this, let's use a separate issue, so that this pull can be merged.

Reply:
I think so. When there's a spec you'll nail down what a running task (as it applies to `maxAsync`) is and what a queued task is.

I think my question comes out of not understanding your model of execution. As you described it, having `maxQueued` doesn't make sense.

However, it's important that these concepts are well defined and explained. Reading the description, I thought "if I have `maxAsync=10` and `maxQueued=10` then I can submit 20 tasks and none should be rejected. However, from your description that isn't true. Just because there's less than `maxAsync` currently running, doesn't mean a newly added task will start immediately.

Reply:
Thank you for clarifying further.  I think I understand now what the problem is.  You are reading the description of maxAsync as though it is a minimum constraint (tasks must start if less than this value) rather than as a maximum.  I have added a commit to update the maxAsync JavaDoc to explicitly state that it does not guarantee tasks will start immediately and to replace the language about running them subject to available capacity (which you probably interpreted as referring to the executor's maxAsync value rather than the system as a whole which was the intended meaning).